eng/pipelines/templates/stages/archetype-js-release.yml

```diff
@@ -20,251 +20,243 @@ stages:
-                                Registry: ${{parameters.Registry}}
-                                PathToArtifacts: $(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.name}}
-                                Tag: "$(Tag)"
-                                AdditionalTag: "$(AdditionalTag)"
-                            - pwsh: >
-                                write-host "$(Package.Archive)"
+                    steps:
+                      - checkout: self
+                      - download: current
+                      - script: |
+                          export DETECTED_PACKAGE_NAME=`ls $(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.name}}/*.tgz`
+                          echo "##vso[task.setvariable variable=Package.Archive;isOutput=true]$DETECTED_PACKAGE_NAME"
+                        name: SetPackageArchive
+                        displayName: Detecting package archive
+                      - pwsh: |
+                          write-host '$(SetPackageArchive.Package.Archive)'
+                          $result = eng/scripts/get-npm-tags.ps1 -packageArtifact '$(SetPackageArchive.Package.Archive)' -workingDirectory '$(System.DefaultWorkingDirectory)/temp'
+                          write-host "Tag: $($result.Tag)"
+                          write-host "Additional tag: $($result.AdditionalTag)"
+                          echo "##vso[task.setvariable variable=Tag;isOutput=true]$($result.Tag)"
+                          echo "##vso[task.setvariable variable=AdditionalTag;isOutput=true]$($result.AdditionalTag)"
+                        condition: and(succeeded(), ne(variables['Skip.AutoAddTag'], 'true'))
+                        name: SetTag
+                        displayName: Set Tag and Additional Tag
+                      - script: >
+                          npm install $(SetPackageArchive.Package.Archive)
+                        displayName: Validating package can be installed
+                        condition: succeeded()
+                  - template: /eng/pipelines/templates/steps/npm-release-task.yml
+                    parameters:
+                      Artifact: ${{artifact}}
+                      Registry: ${{parameters.Registry}}
+                      PathToArtifacts: $(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.name}}
+                      Tag: "$(Tag)"
+                      AdditionalTag: "$(AdditionalTag)"
+                      DependsOn:
+                        - 'PrePublishPackage'
+                      VarDependencies:
+                        Tag: $[dependencies.PrePublishPackage.outputs['SetTag.Tag']]
+                        AdditionalTag: $[dependencies.PrePublishPackage.outputs['SetTag.AdditionalTag']]
+                  - job: PostPublishPackage
+                    dependsOn: PublishPackage
+                    variables:
+                      'Package.Archive': $[dependencies.PrePublishPackage.outputs['SetPackageArchive.Package.Archive']]
```

Comment:
Are you actually setting `SetPackageArchive.Package.Archive` anywhere?

Reply:
yeah we need that for eng/scripts/cleanup-npm-next-tag.ps1